apps/dashboard/src/pages/navbar-pages/staking/validator-details/ApyDetail.svelte

```diff
@@ -0,0 +1,56 @@
+<script lang="ts">
+  import { type UptimeValue } from '@api/utils/entities/component/validator'
+  import SkeletonLoader from '@components/_base/skeleton-loader/SkeletonLoader.svelte'
+  import { uptimeModule } from '@dashboard/lib/validators/uptime-module'
+  import { truncateNumber } from '@utils'
+  import { onMount } from 'svelte'
+  import type { TransformedValidator } from '../validator-list/ValidatorList.svelte'
+
+  export let validator: {
+    typed: {
+      type: 'String'
+      value: TransformedValidator
+    }
+  }
+
+  let selected: UptimeValue = '1month'
+  let isLoading: boolean
+  let validatorsUptimeData = uptimeModule.getDataForUptime(selected)
+
+  onMount(() => {
+    const subscription = uptimeModule.isLoading$.subscribe((data) => {
+      isLoading = data
+      validatorsUptimeData = uptimeModule.getDataForUptime(selected)
+    })
+
+    return () => {
+      subscription.unsubscribe()
+    }
+  })
+  $: {
+    validatorsUptimeData = uptimeModule.getDataForUptime(selected)
+    isLoading
+  }
+</script>
+
+{#key isLoading}
```

Comment:
I don't see why `isLoading` is necessary?

it seems that `validatorsUptimeData` will change and then the ui should rerender based on it

Reply:
It's not re-rendered automatically 🤷‍♂️ I think it's because `validatorsUptimeData` has the same object where just internals change or maybe it's because it's done out of svelte file context? Either way it doesn't work without it and I had to use this pattern a bunch of times already

Reply:
But UI was not updated, I can double check but I'm pretty sure about that

Reply:
it was updated for me